astropy/coordinates/earth.py

```diff
@@ -595,14 +593,28 @@ def to_geodetic(self, ellipsoid=None):
         For the conversion to geodetic coordinates, the ERFA routine
         ``gc2gd`` is used.  See https://github.com/liberfa/erfa
         """
+        warn("Method .to_geodetic(ellipsoid) is deprecated, "
+             "use .represent_as({WGS84,WGS72,GRS80}GeodeticRepresentation) instead ",
+             AstropyDeprecationWarning)
+
         ellipsoid = _check_ellipsoid(ellipsoid, default=self.ellipsoid)
-        self_array = self.to(u.meter).view(self._array_dtype, np.ndarray)
-        lon, lat, height = erfa.gc2gd(getattr(erfa, ellipsoid), self_array)
+        llh = CartesianRepresentation(self.x, self.y, self.z).represent_as(
+            REPRESENTATION_CLASSES[ellipsoid.lower() + "geodetic"])
         return GeodeticLocation(
-            Longitude(lon * u.radian, u.degree,
-                      wrap_angle=180.*u.degree, copy=False),
-            Latitude(lat * u.radian, u.degree, copy=False),
-            u.Quantity(height * u.meter, self.unit, copy=False))
+            llh.lon,
+            llh.lat,
+            u.Quantity(llh.height, self.unit, copy=False))
+
+    def represent_as(self, other_class):
```

Comment:
Again seeing `EarthLocation` as more like a frame/skycoord rather than a representation, I think we can allow strings. I'd suggest following the nomenclature in their `represent_as` method and calling it `base`.

Reply:
Removed `represent_as` for now, but saving this for the future.